haskeline.cabal

```diff
@@ -108,4 +108,23 @@ Library
             cpp-options: -DUSE_TERMIOS_H
         }
     }
-    ghc-options: -Wall
+
+Executable tests
+    hs-source-dirs: tests
+    Default-Language: Haskell98
+
+    if os(windows)
+        -- this is dummy module, tests don't compile on windows
+        Main-Is: Windows.hs
+        Build-depends: base
+    else
+        Main-Is:    Unit.hs
+        Build-depends: base, containers, text, bytestring, HUnit, process, unix
+        Other-Modules: RunTTY, Pty
+
+-- The following program is used by unit tests in `tests` executable
+Executable examples-Test
```

Comment:
> Add a Cabal flag, off by default, that guards whether these executables are buildable

btw, also make sure to declare it a "manual" flag, otherwise there's no guarantee it won't be toggled on automatically by the solver

Reply:
I understand no one would want those programs on their path, but why would anyone call "cabal install haskeline" in the first place? This is a library that one adds as a dependency.

Reply:
I prefixed the executables names.